tchannel-core/src/main/java/com/uber/tchannel/tracing/OpenTracingContext.java

```diff
@@ -52,42 +52,46 @@ public OpenTracingContext(@NotNull ScopeManager scopeManager) {
     @Override
     @SuppressWarnings("resource")
     public void pushSpan(@NotNull Span span) {
-        scopeManager.activate(span, false);
+        scopeManager.activate(span);
     }
 
     @Override
     @SuppressWarnings("resource")
     public boolean hasSpan() {
-        return scopeManager.active() != null;
+        return scopeManager.activeSpan() != null;
     }
 
     @Override
     public @NotNull Span currentSpan() throws EmptyStackException {
-        @SuppressWarnings("resource") Scope scope = scopeManager.active();
-        if (scope == null) {
+        Span span = scopeManager.activeSpan();
+        if (span == null) {
             throw new EmptyStackException();
-        } else {
-            return scope.span();
         }
+        return span;
     }
 
     @Override
     public @NotNull Span popSpan() throws EmptyStackException {
+        //auto close scope acts as if we pop the span
         try (Scope scope = scopeManager.active()) {
             if (scope == null) {
                 throw new EmptyStackException();
```

Comment:
what is going to happen to `try` with null?

Reply:
A resource is closed only if it initialized to a non-null value. When scope is null, it won't be closed and there won't be NPE after the EmptyStackException is thrown. There is unit test that covers this scenario.